compiler/qsc_frontend/src/error.rs

```diff
@@ -70,15 +72,16 @@ impl<E: Diagnostic + Send + Sync> WithSource<E> {
     /// Takes a span that uses `SourceMap` offsets, and returns
     /// a span that is relative to the `Source` that the span falls into,
     /// along with a reference to the `Source`.
-    pub fn resolve_span(&self, span: &SourceSpan) -> (&Source, SourceSpan) {
+    pub fn resolve_span(&self, span: &SourceSpan) -> Result<(&Source, SourceSpan), MietteError> {
```

Comment:
I'm worried about this. Is there really a legitimate case where the span can get out of sync with the sources contained in the error?

Reply:
it's not that the span gets out sync, but rather that there is no source code for the line the call originates from. But with some more testing, it appears that this is specific to invoking stdlib intrinsic callables directly from Python. Since these changes don't support that, I can take this and the related code out and it's one less oddity to worry about in this PR. That said, if we later need to support Python calls into intrinsics from the stdlib, we might need to revisit this. Longer explanation below...

When an intrinsic callable fails internally it can't point to those internals so it instead uses the span of the call expression in the error. But if the line that actually invoked that callable was in Python, then there is no source code for that call expression and thus no span. I ran into this when I tried manually calling `__quantum__rt__qubit_release` on a qubit in the non-zero state, because it uses the span from the arguments of the call expression:
https://github.com/microsoft/qsharp/blob/f02677479532caa2c252ab58b6d4e15a7247fc0a/compiler/qsc_eval/src/lib.rs#L1130
Since the original call was from a line of Python like, `qsharp.env.QIR.Runtime.__quantum__rt__qubit_release(q)`, there is no Q# source for that argument span. This helper would panic while trying to create the error to report.

Again, all that is moot now, so I'll remove this part of the change. But we should keep this in the back of our minds if we ever open up the ability to directly invoke standard library intrinsics.

Reply:
That's good to know thanks for explaining! When the time comes we can discuss how to handle the case. In the meantime I'll stop worrying :)

Reply:
Good call out, though! Code is removed now.